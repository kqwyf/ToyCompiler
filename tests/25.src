int main() {
    bool a, b;
    a = 1 == 1;
    b = 1 == 1;
    if(a && (1 == 1)) {
        b = 1 == 0;
    }
    if((a == b) || b) {
        a = 1 == 0;
    }
    if((1 == 1) && (1 == 0)) {
        a = 1 == 1;
    }
    if(a || b) {
        b = 1 == 0;
    }
    if(!a) {
        b = 1 == 1;
    }
    if(!(a != b)) {
        a = 1 == 0;
    }
}
