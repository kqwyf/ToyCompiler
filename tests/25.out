Token sequence:
int             < INT         ,        >
main            < IDENTIFIER  , 1      >
(               < LEFTPAREN   ,        >
)               < RIGHTPAREN  ,        >
{               < LEFTBRACE   ,        >
bool            < BOOL        ,        >
a               < IDENTIFIER  , 2      >
,               < COMMA       ,        >
b               < IDENTIFIER  , 3      >
;               < SEMICOLON   ,        >
a               < IDENTIFIER  , 2      >
=               < ASSIGN      ,        >
1               < CONSTANT    , 4      >
==              < EQUAL       ,        >
1               < CONSTANT    , 4      >
;               < SEMICOLON   ,        >
b               < IDENTIFIER  , 3      >
=               < ASSIGN      ,        >
1               < CONSTANT    , 4      >
==              < EQUAL       ,        >
1               < CONSTANT    , 4      >
;               < SEMICOLON   ,        >
if              < IF          ,        >
(               < LEFTPAREN   ,        >
a               < IDENTIFIER  , 2      >
&&              < AND         ,        >
(               < LEFTPAREN   ,        >
1               < CONSTANT    , 4      >
==              < EQUAL       ,        >
1               < CONSTANT    , 4      >
)               < RIGHTPAREN  ,        >
)               < RIGHTPAREN  ,        >
{               < LEFTBRACE   ,        >
b               < IDENTIFIER  , 3      >
=               < ASSIGN      ,        >
1               < CONSTANT    , 4      >
==              < EQUAL       ,        >
0               < CONSTANT    , 5      >
;               < SEMICOLON   ,        >
}               < RIGHTBRACE  ,        >
if              < IF          ,        >
(               < LEFTPAREN   ,        >
(               < LEFTPAREN   ,        >
a               < IDENTIFIER  , 2      >
==              < EQUAL       ,        >
b               < IDENTIFIER  , 3      >
)               < RIGHTPAREN  ,        >
||              < OR          ,        >
b               < IDENTIFIER  , 3      >
)               < RIGHTPAREN  ,        >
{               < LEFTBRACE   ,        >
a               < IDENTIFIER  , 2      >
=               < ASSIGN      ,        >
1               < CONSTANT    , 4      >
==              < EQUAL       ,        >
0               < CONSTANT    , 5      >
;               < SEMICOLON   ,        >
}               < RIGHTBRACE  ,        >
if              < IF          ,        >
(               < LEFTPAREN   ,        >
(               < LEFTPAREN   ,        >
1               < CONSTANT    , 4      >
==              < EQUAL       ,        >
1               < CONSTANT    , 4      >
)               < RIGHTPAREN  ,        >
&&              < AND         ,        >
(               < LEFTPAREN   ,        >
1               < CONSTANT    , 4      >
==              < EQUAL       ,        >
0               < CONSTANT    , 5      >
)               < RIGHTPAREN  ,        >
)               < RIGHTPAREN  ,        >
{               < LEFTBRACE   ,        >
a               < IDENTIFIER  , 2      >
=               < ASSIGN      ,        >
1               < CONSTANT    , 4      >
==              < EQUAL       ,        >
1               < CONSTANT    , 4      >
;               < SEMICOLON   ,        >
}               < RIGHTBRACE  ,        >
if              < IF          ,        >
(               < LEFTPAREN   ,        >
a               < IDENTIFIER  , 2      >
||              < OR          ,        >
b               < IDENTIFIER  , 3      >
)               < RIGHTPAREN  ,        >
{               < LEFTBRACE   ,        >
b               < IDENTIFIER  , 3      >
=               < ASSIGN      ,        >
1               < CONSTANT    , 4      >
==              < EQUAL       ,        >
0               < CONSTANT    , 5      >
;               < SEMICOLON   ,        >
}               < RIGHTBRACE  ,        >
if              < IF          ,        >
(               < LEFTPAREN   ,        >
!               < NOT         ,        >
a               < IDENTIFIER  , 2      >
)               < RIGHTPAREN  ,        >
{               < LEFTBRACE   ,        >
b               < IDENTIFIER  , 3      >
=               < ASSIGN      ,        >
1               < CONSTANT    , 4      >
==              < EQUAL       ,        >
1               < CONSTANT    , 4      >
;               < SEMICOLON   ,        >
}               < RIGHTBRACE  ,        >
if              < IF          ,        >
(               < LEFTPAREN   ,        >
!               < NOT         ,        >
(               < LEFTPAREN   ,        >
a               < IDENTIFIER  , 2      >
!=              < NOTEQUAL    ,        >
b               < IDENTIFIER  , 3      >
)               < RIGHTPAREN  ,        >
)               < RIGHTPAREN  ,        >
{               < LEFTBRACE   ,        >
a               < IDENTIFIER  , 2      >
=               < ASSIGN      ,        >
1               < CONSTANT    , 4      >
==              < EQUAL       ,        >
0               < CONSTANT    , 5      >
;               < SEMICOLON   ,        >
}               < RIGHTBRACE  ,        >
}               < RIGHTBRACE  ,        >

Symbol table:
1     main
2     a
3     b
4     1
5     0

Production sequence:
TYPE_BASIC -> int
TYPE -> TYPE_BASIC
DECLARE_FUNC_BEGIN -> TYPE identifier (
DECLARE_FUNC_MID -> ) {
TYPE_BASIC -> bool
TYPE -> TYPE_BASIC
IDENTIFIER_S -> identifier
IDENTIFIER_S -> IDENTIFIER_S , identifier
DECLARE_VAR -> TYPE IDENTIFIER_S ;
STATEMENT -> DECLARE_VAR
STATEMENT_S -> STATEMENT
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION8 -> EXPRESSION7 = EXPRESSION8
EXPRESSION -> EXPRESSION8
STATEMENT -> EXPRESSION ;
STATEMENT_S -> STATEMENT_S STATEMENT
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION8 -> EXPRESSION7 = EXPRESSION8
EXPRESSION -> EXPRESSION8
STATEMENT -> EXPRESSION ;
STATEMENT_S -> STATEMENT_S STATEMENT
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION -> EXPRESSION8
EXPRESSION2 -> ( EXPRESSION )
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION7 && EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION -> EXPRESSION8
SELECT_BEGIN -> if ( EXPRESSION )
STATEMENTS_BEGIN -> {
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION8 -> EXPRESSION7 = EXPRESSION8
EXPRESSION -> EXPRESSION8
STATEMENT -> EXPRESSION ;
STATEMENT_S -> STATEMENT
STATEMENT -> STATEMENTS_BEGIN STATEMENT_S }
SELECT -> SELECT_BEGIN STATEMENT
STATEMENT -> SELECT
STATEMENT_S -> STATEMENT_S STATEMENT
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION -> EXPRESSION8
EXPRESSION2 -> ( EXPRESSION )
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION7 || EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION -> EXPRESSION8
SELECT_BEGIN -> if ( EXPRESSION )
STATEMENTS_BEGIN -> {
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION8 -> EXPRESSION7 = EXPRESSION8
EXPRESSION -> EXPRESSION8
STATEMENT -> EXPRESSION ;
STATEMENT_S -> STATEMENT
STATEMENT -> STATEMENTS_BEGIN STATEMENT_S }
SELECT -> SELECT_BEGIN STATEMENT
STATEMENT -> SELECT
STATEMENT_S -> STATEMENT_S STATEMENT
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION -> EXPRESSION8
EXPRESSION2 -> ( EXPRESSION )
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION -> EXPRESSION8
EXPRESSION2 -> ( EXPRESSION )
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION7 && EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION -> EXPRESSION8
SELECT_BEGIN -> if ( EXPRESSION )
STATEMENTS_BEGIN -> {
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION8 -> EXPRESSION7 = EXPRESSION8
EXPRESSION -> EXPRESSION8
STATEMENT -> EXPRESSION ;
STATEMENT_S -> STATEMENT
STATEMENT -> STATEMENTS_BEGIN STATEMENT_S }
SELECT -> SELECT_BEGIN STATEMENT
STATEMENT -> SELECT
STATEMENT_S -> STATEMENT_S STATEMENT
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION7 || EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION -> EXPRESSION8
SELECT_BEGIN -> if ( EXPRESSION )
STATEMENTS_BEGIN -> {
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION8 -> EXPRESSION7 = EXPRESSION8
EXPRESSION -> EXPRESSION8
STATEMENT -> EXPRESSION ;
STATEMENT_S -> STATEMENT
STATEMENT -> STATEMENTS_BEGIN STATEMENT_S }
SELECT -> SELECT_BEGIN STATEMENT
STATEMENT -> SELECT
STATEMENT_S -> STATEMENT_S STATEMENT
EXPRESSION2 -> identifier
EXPRESSION2 -> ! EXPRESSION2
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION -> EXPRESSION8
SELECT_BEGIN -> if ( EXPRESSION )
STATEMENTS_BEGIN -> {
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION8 -> EXPRESSION7 = EXPRESSION8
EXPRESSION -> EXPRESSION8
STATEMENT -> EXPRESSION ;
STATEMENT_S -> STATEMENT
STATEMENT -> STATEMENTS_BEGIN STATEMENT_S }
SELECT -> SELECT_BEGIN STATEMENT
STATEMENT -> SELECT
STATEMENT_S -> STATEMENT_S STATEMENT
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 != EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION -> EXPRESSION8
EXPRESSION2 -> ( EXPRESSION )
EXPRESSION2 -> ! EXPRESSION2
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION -> EXPRESSION8
SELECT_BEGIN -> if ( EXPRESSION )
STATEMENTS_BEGIN -> {
EXPRESSION2 -> identifier
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION5
EXPRESSION2 -> constant
EXPRESSION3 -> EXPRESSION2
EXPRESSION4 -> EXPRESSION3
EXPRESSION5 -> EXPRESSION4
EXPRESSION6 -> EXPRESSION6 == EXPRESSION5
EXPRESSION7 -> EXPRESSION6
EXPRESSION8 -> EXPRESSION7
EXPRESSION8 -> EXPRESSION7 = EXPRESSION8
EXPRESSION -> EXPRESSION8
STATEMENT -> EXPRESSION ;
STATEMENT_S -> STATEMENT
STATEMENT -> STATEMENTS_BEGIN STATEMENT_S }
SELECT -> SELECT_BEGIN STATEMENT
STATEMENT -> SELECT
STATEMENT_S -> STATEMENT_S STATEMENT
DECLARE_FUNC -> DECLARE_FUNC_BEGIN DECLARE_FUNC_MID STATEMENT_S }
DECLARE -> DECLARE_FUNC
DECLARE_S -> DECLARE
PROGRAM -> DECLARE_S

Semantic Symbol Tables:

Global Symbol Table (Table 0):
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------
    0 | INT          | 1            | 0        | 
    1 | INT          | 0            | 4        | 
    2 | BLOCK        | main         | .L42     | Table 1, Params#: 0

Table 1:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------
    0 | INT          | (anonymous)  | 0        | 
    1 | BOOL         | a            | 4        | 
    2 | BOOL         | b            | 5        | 
    3 | BLOCK        | (anonymous)  |          | Table 2
    4 | BLOCK        | (anonymous)  |          | Table 4
    5 | BLOCK        | (anonymous)  |          | Table 6
    6 | BLOCK        | (anonymous)  |          | Table 8
    7 | BLOCK        | (anonymous)  |          | Table 10
    8 | BLOCK        | (anonymous)  |          | Table 12

Table 2:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------
    0 | BLOCK        | (anonymous)  |          | Table 3

Table 3:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------

Table 4:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------
    0 | BLOCK        | (anonymous)  |          | Table 5

Table 5:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------

Table 6:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------
    0 | BLOCK        | (anonymous)  |          | Table 7

Table 7:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------

Table 8:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------
    0 | BLOCK        | (anonymous)  |          | Table 9

Table 9:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------

Table 10:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------
    0 | BLOCK        | (anonymous)  |          | Table 11

Table 11:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------

Table 12:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------
    0 | BLOCK        | (anonymous)  |          | Table 13

Table 13:
  #   |   DataType   |     Name     |  Offset  |    Attr
------+--------------+--------------+----------+------------


Instruction sequence:
.L42   (  JE,   0:0  ,   0:0  , .L1    )
       ( JMP,        ,        , .L2    )
.L1    ( TRU,        ,        ,   1:3  )
       ( JMP,        ,        , .L0    )
.L2    ( FAL,        ,        ,   1:3  )
.L0    ( MOV,   1:1  ,   1:3  ,        )
       ( JNZ,   1:1  ,        , .L6    )
       ( JMP,        ,        , .L6    )
.L6    (  JE,   0:0  ,   0:0  , .L4    )
       ( JMP,        ,        , .L5    )
.L4    ( TRU,        ,        ,   1:3  )
       ( JMP,        ,        , .L3    )
.L5    ( FAL,        ,        ,   1:3  )
.L3    ( MOV,   1:2  ,   1:3  ,        )
       ( JNZ,   1:2  ,        , .L12   )
       ( JMP,        ,        , .L12   )
.L12   ( JNZ,   1:1  ,        , .L7    )
       ( JMP,        ,        , .L18   )
.L7    (  JE,   0:0  ,   0:0  , .L11   )
       ( JMP,        ,        , .L18   )
.L11   (  JE,   0:0  ,   0:1  , .L9    )
       ( JMP,        ,        , .L10   )
.L9    ( TRU,        ,        ,   3:0  )
       ( JMP,        ,        , .L8    )
.L10   ( FAL,        ,        ,   3:0  )
.L8    ( MOV,   1:2  ,   3:0  ,        )
       ( JNZ,   1:2  ,        , .L18   )
       ( JMP,        ,        , .L18   )
.L18   (  JE,   1:1  ,   1:2  , .L17   )
       ( JMP,        ,        , .L13   )
.L13   ( JNZ,   1:2  ,        , .L17   )
       ( JMP,        ,        , .L24   )
.L17   (  JE,   0:0  ,   0:1  , .L15   )
       ( JMP,        ,        , .L16   )
.L15   ( TRU,        ,        ,   5:0  )
       ( JMP,        ,        , .L14   )
.L16   ( FAL,        ,        ,   5:0  )
.L14   ( MOV,   1:1  ,   5:0  ,        )
       ( JNZ,   1:1  ,        , .L24   )
       ( JMP,        ,        , .L24   )
.L24   (  JE,   0:0  ,   0:0  , .L19   )
       ( JMP,        ,        , .L30   )
.L19   (  JE,   0:0  ,   0:1  , .L23   )
       ( JMP,        ,        , .L30   )
.L23   (  JE,   0:0  ,   0:0  , .L21   )
       ( JMP,        ,        , .L22   )
.L21   ( TRU,        ,        ,   7:0  )
       ( JMP,        ,        , .L20   )
.L22   ( FAL,        ,        ,   7:0  )
.L20   ( MOV,   1:1  ,   7:0  ,        )
       ( JNZ,   1:1  ,        , .L30   )
       ( JMP,        ,        , .L30   )
.L30   ( JNZ,   1:1  ,        , .L29   )
       ( JMP,        ,        , .L25   )
.L25   ( JNZ,   1:2  ,        , .L29   )
       ( JMP,        ,        , .L35   )
.L29   (  JE,   0:0  ,   0:1  , .L27   )
       ( JMP,        ,        , .L28   )
.L27   ( TRU,        ,        ,   9:0  )
       ( JMP,        ,        , .L26   )
.L28   ( FAL,        ,        ,   9:0  )
.L26   ( MOV,   1:2  ,   9:0  ,        )
       ( JNZ,   1:2  ,        , .L35   )
       ( JMP,        ,        , .L35   )
.L35   ( JNZ,   1:1  ,        , .L40   )
       ( JMP,        ,        , .L34   )
.L34   (  JE,   0:0  ,   0:0  , .L32   )
       ( JMP,        ,        , .L33   )
.L32   ( TRU,        ,        ,  11:0  )
       ( JMP,        ,        , .L31   )
.L33   ( FAL,        ,        ,  11:0  )
.L31   ( MOV,   1:2  ,  11:0  ,        )
       ( JNZ,   1:2  ,        , .L40   )
       ( JMP,        ,        , .L40   )
.L40   ( JNE,   1:1  ,   1:2  , .L41   )
       ( JMP,        ,        , .L39   )
.L39   (  JE,   0:0  ,   0:1  , .L37   )
       ( JMP,        ,        , .L38   )
.L37   ( TRU,        ,        ,  13:0  )
       ( JMP,        ,        , .L36   )
.L38   ( FAL,        ,        ,  13:0  )
.L36   ( MOV,   1:1  ,  13:0  ,        )
       ( JNZ,   1:1  ,        , .L41   )
       ( JMP,        ,        , .L41   )
.L41   ( RET,        ,        ,        )
